server/application/application.go

```diff
@@ -206,6 +211,33 @@ func (s *Server) Create(ctx context.Context, q *application.ApplicationCreateReq
 	return updated, nil
 }
 
+// ListNodes returns nodes associated with an application
+func (s *Server) ListNodes(ctx context.Context, q *application.NodeQuery) (*v1.NodeList, error) {
+	nodes, err := s.nodeLister.List(labels.Everything())
```

Comment:
Nodes list should be filtered and API should return only nodes that are referenced by at least one application resource ( returned by `tree, err := s.getAppResources(ctx, a)`  )